modules/library/main/src/main/java/org/geotools/styling/visitor/RenderingSelectorStyleVisitor.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+
+package org.geotools.styling.visitor;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.geotools.styling.FeatureTypeStyle;
+import org.geotools.styling.Rule;
+import org.geotools.styling.Symbolizer;
+
+/**
+ * This class applies the renderingLegend and renderingMap VendorOptions to a Style eg.
+ * <VendorOption name="renderingLegend">false</VendorOption> <VendorOption
+ * name="renderingMap">false</VendorOption> The visitor avoid including in the deep copy of the
+ * Style being passed to it the FeatureTypeStyle or the Rule or the Symbolizer for which the
+ * vendorOptions passed to the constructor has been defined with a false value.
+ */
+public class RenderingSelectorStyleVisitor extends DuplicatingStyleVisitor {
+
+    public static final String RENDERING_MAP_OPTION = "renderingMap";
+    public static final String RENDERING_LEGEND_OPTION = "renderingLegend";
+
+    private static final List<String> SUPPORTED_OPTIONS =
+            Arrays.asList(RENDERING_LEGEND_OPTION, RENDERING_MAP_OPTION);
+    private String renderingOption;
+
+    public RenderingSelectorStyleVisitor(String renderingOption) {
+        if (!SUPPORTED_OPTIONS.contains(renderingOption))
+            throw new UnsupportedOperationException(
+                    "The RenderingSelector visitor is meant to work with the following VendorOptions: "
+                            + RENDERING_MAP_OPTION
```

Comment:
Can just use the SUPPORTED_OPTIONS list again.

Reply:
this changed after having applied the feedback on the devel list